metadata_store_scripts/V2025.07.22.0__add_generate_masking_parameters_procedure.sql

```diff
@@ -0,0 +1,391 @@
+                CASE
+                    WHEN
+                        CEILING(
+                            MAX(row_count)
+                            * (
+                                SUM(column_width_estimate)
+                                + LOG10(MAX(row_count))
+                                + 1
+                            )
+                            / (2000000 * 0.9)
+                        ) < 1
+                        THEN 1
+                    ELSE CEILING(
+                        MAX(row_count)
+                        * (
+                            SUM(column_width_estimate)
+                            + LOG10(MAX(row_count))
+                            + 1
+                        )
+                        / (2000000 * 0.9)
+                    )
+                END, 1
+            ) AS [NumberOfBatches],   -- Optimal batch count (minimum 1)
+            COALESCE(
+                '[' + STRING_AGG(
+                    CONVERT(NVARCHAR(10), identified_column_max_length), ','
+                ) + ']', '[]'
+            ) AS [TrimLengths]   -- JSON array of column max lengths
+        FROM ruleset_with_types
+    ),
+
+    -- conditional_date_formats - Extract conditional date formats for filter key
+    conditional_date_formats AS (
+        SELECT
+            f.identified_column,
+            f.encoded_column_name,
+            c.[condition_alias],
+            c.[date_format] AS conditional_date_format
+        FROM ruleset_computed AS f
+        CROSS APPLY
+            OPENJSON (f.algorithm_metadata, '$.conditions')
+            WITH (
+                [condition_alias] NVARCHAR(255) '$.alias',
+                [date_format] NVARCHAR(255) '$.date_format'
+            ) AS c
+        WHERE
+            @filter_key <> ''
+            AND f.algorithm_metadata IS NOT NULL
+            AND c.[condition_alias] = @filter_key
+            AND c.[date_format] IS NOT NULL
+    ),
+
+    -- date_format_resolution - Merge conditional and standard date formats
+    date_format_resolution AS (
+        SELECT
+            f.identified_column,
+            f.encoded_column_name,
+            COALESCE(cdf.conditional_date_format, f.[date_format]) AS final_date_format
+        FROM ruleset_computed AS f
+        LEFT JOIN conditional_date_formats AS cdf
+            ON f.identified_column = cdf.identified_column
+        WHERE
+            f.[date_format] IS NOT NULL
+            OR cdf.conditional_date_format IS NOT NULL
+    ),
+
+    -- date_format_parameters - Create JSON mapping with fallback default
+    date_format_parameters AS (
+        SELECT
+            COALESCE(
+                '{' + STRING_AGG(
+                    '"' + LOWER(dfr.encoded_column_name) + '":"' + dfr.final_date_format + '"',
+                    ','
+                ) + '}',
+                '{}'
+            ) AS [DateFormatAssignments]
+        FROM date_format_resolution AS dfr
+        WHERE dfr.final_date_format IS NOT NULL
```

Comment:
note: `final_date_format` can't be null, since it's determined by a `COALESCE` on two things that are part of a selection criteria where either of them is required to be non-null